eth/client.go

```diff
@@ -738,8 +739,99 @@ func (c *client) Vote(pollAddr ethcommon.Address, choiceID *big.Int) (*types.Tra
 	return poll.Vote(opts, choiceID)
 }
 
+func (c *client) Reward() (*types.Transaction, error) {
+	addr := c.accountManager.Account().Address
+
+	t, err := c.GetTranscoder(addr)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder %v", addr.Hex())
+	}
+
+	mintable, err := c.CurrentMintableTokens()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get curernt mintable tokens")
+	}
+
+	totalBonded, err := c.GetTotalBonded()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get total bonded")
+	}
+
+	// reward = (current mintable tokens for the round * active transcoder stake) / total active stake
+	reward := new(big.Int).Div(new(big.Int).Mul(mintable, t.DelegatedStake), totalBonded)
+
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get transcoder pool")
+	}
+
+	hints := c.getStakingHints(addr, ethcommon.Address{}, reward, transcoders)
+	return c.RewardWithHint(hints.newPosPrev, hints.newPosNext)
+
+}
+
 // Helpers
 
+// getStakingHints returns hints for both the new and old (if applicable) delegate
+// amount provided should be negative when unbonding.
+func (c *client) getStakingHints(newDel ethcommon.Address, oldDel ethcommon.Address, amount *big.Int, transcoders []*lpTypes.Transcoder) *struct {
```

Comment:
Looks like this function does more than just get a hint since it actually needs to simulate an update (mutate a slice) to the transcoder pool in order to calculate the hint. I think something like `simulateTranscoderPoolUpdate()` would be more descriptive.